share/ci/scripts/linux/run_gcov.sh

```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env bash
+
+set -ex
+
+mkdir _coverage
+cd _coverage
+
+# The sed command below converts from:
+#   ../_build/src/OpenColorIO/CMakeFiles/OpenColorIO.dir/ops/Exponent.gcno
+# to:
+#   ../src/OpenColorIO/ops/Exponent.cpp
+
+for g in $(find ../_build -name "*.gcno" -type f); do
+    gcov -p -o $(dirname "$g") $(echo "$g" | sed -e 's/\/_build\//\//' -e 's/\.gcno/\.cpp/' -e 's/\/CMakeFiles.*\.dir\//\//')
```

Comment:
@michdolan - I've been experimenting with SonarCloud recently and copied some of what you're doing here (thanks!). One issue I ran into is that the code coverage for header files didn't seem correct. I believe the issue was because many of our headers were being #included in more than one translation unit, so the gcov files for the headers were being overwritten.

Long story short, including the --long-file-names (-l) flag to gcov solved the problem because it prefixed the .cpp filename, hence making the header gcov files unique. See:

https://gcc.gnu.org/onlinedocs/gcc/Invoking-Gcov.html

Mentioning this in case it's useful here (though I'll point out that I've been mainly using clang + llvm-cov gcov so might not be relevant).

Reply:
Thanks @danrbailey ! Good catch. I'll give that a try.